src/createXcmTypes/util/foreignAssetMultiLocationIsInRegistry.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+
+import { ApiPromise } from '@polkadot/api';
+
+import { BaseError } from '../../errors';
+import { Registry } from '../../registry';
+import { ForeignAssetsInfo } from '../../registry/types';
+// import { ForeignAssetMultiLocation } from "src/types";
+
+export const foreignAssetMultiLocationIsInRegistry = (
+	multilocationStr: string,
+	registry: Registry,
+	api: ApiPromise
+): boolean => {
+	try {
+		const assetHubChainId = 1000;
+		const multiLocation = api.registry.createType(
```

Comment:
Would this have to be in a try-catch since it will throw an error if string is incorrect, or do we do error validation on this inside of checkXcmTxInput so it wont happen?

Reply:
Great question! This is in a try catch since if the `multilocation` string is invalid it will throw an error. This same function is used in both `checkLocalTxInput` and `checkXcmTxInput` to check if a `multilocation` asset exists and will error in either of them if the string is invalid. It's also being used in `getSystemChainAssetId` for local txs.

Reply:
Awesome :), Thanks for the explanation.